src/test/java/racegame/domain/WinnersTest.java

```diff
@@ -0,0 +1,20 @@
+package racegame.domain;
+
+import static org.assertj.core.api.Assertions.*;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+
+public class WinnersTest {
+
+    @Test
+    public void winnersCreateTest() {
+        Car pobi = new Car("pobi", 5);
+        Car teo = new Car("teo", 8);
+        Car tarkov = new Car("tarko", 8);
+        Cars cars = new Cars(Arrays.asList(pobi, teo, tarkov));
+        Winners winners = cars.getWinners();
+        assertThat(winners.toString()).isEqualTo("teo, tarko");
+    }
+
+}
```

Comment:
### 1. 리팩터링 작업을 할 때, 커밋 시점은 어떻게 잡아야 할까요? 컴파일 에러가 없는 상태에서 커밋하려고 하다보니 사실상 리팩터링 다 끝나고 커밋해버렸습니다. 😱

어느정도 편하신 시점에 하시면 됩니다. 다만 커밋 이력을 확인해보니 말씀처럼 한번에 다 커밋하셨더군요. 😭 작업이 작으면 한번에도 하긴 하는데 그래도 
- 테스트 코드를 작성 완료
- 구현 완료
- 리팩토링 완료 - 복잡한 기능이 아니였다면 위 단계에 포함시킵니다.

작은 기능을 작업할 때 케이스고 teo가 진행하신 리팩토링이였다면 
- 자잘한(?) 리뷰 반영
- Position, Name 적용
- 일급 컬렉션 적용
- 위 적용 사항에 맞춰 전체적인 리팩토링

이정도로 분리해서 작업했을 것 같습니다.

Reply:
### 2. 스트림에 인덴트 규약이 있나요? 온점(.)을 수직선상에서 동일한 위치로 맞춰보았는데, IDE 자동 정렬을 사용하니 특정한 위치로 가더라구요.

이건 저나 팀원 모두 크게 개의치 않아 말씀주신 김에 찾아봤는데 스트림을 직접적으로 언급하여 가이드를 제공하진 않네요. 너무 독특하게(?)만 하지 않는다면 편하신 스타일로 하면 될 것 같습니다. 

사실은
> [2.3.1 Whitespace characters](https://google.github.io/styleguide/javaguide.html#s2.3.1-whitespace-characters)
> 2. Tab characters are not used for indentation.
> [4.1.1 Braces are used where optional](https://google.github.io/styleguide/javaguide.html#s4.1.1-braces-always-used)
Braces are used with if, else, for, do and while statements, even when the body is empty or contains only a single statement.

이 두개만 아니면 싸움날일 없습니다. 특히 탭 😄 
(intellij에선 탭 사용시 공백 2칸인가 4칸으로 기본 설정되어 있습니다) 

참고 가이드 :
원문 [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html)
한글 [[Java] 35. 코딩 스타일 설정(Google Standard coding style)](https://nowonbun.tistory.com/378)

Reply:
### 3. 자바 문법적으로 궁금한 부분이 있습니다. GameManager 클래스의 makeCars() 메서드는 new Cars( 스트림…) 형태로 되어있어서 마지막에 뭐가 나오는지 알아보기 힘든 것 같습니다. 개선할 수 있는 방법이 있을까요? collect(Collectors.toList()) 로 리스트를 얻어오듯이, 스트림 마지막에 무언가를 붙여서 Cars 객체 형태로 리턴받는 방법이 없을까 궁금합니다.

고민해봤는데 이거다 떠오르는게 없네요. 😭 🙏  혹시 생각나는 방법이 있다면 말씀드릴게요.
(collect이 최종 연산인건 아실것 같아 스트림에 대한 설명은 생략합니다.ㅎㅎ)

Reply:
### 5. 코드리뷰 3단계를 진행하는 과정에서 git rebase를 비롯한 여러 복잡한 절차를 거쳤는데 이것이 왜 필요한지 궁금합니다. git pull 로 한번에 할 수 있는 것 같은데 어떤 차이가 있나요??

복잡한 절차가 아마 fetch와 rebase일 것 같네요.

[Git Merge와 Rebase의 차이, 아름다운고 깔끔한 Git History 만들기.](https://firework-ham.tistory.com/12)
[git의 merge와 rebase 비교하기](https://blog.outsider.ne.kr/666)

```bash
 git pull --help
```
위 명령어를 치면 pull에 대한 설명을 볼 수 있습니다.(깃 버전 등 몇가지 환경에 따라 다른 결과가 나올 수 있습니다.)

> DESCRIPTION
       Incorporates changes from a remote repository into the current branch. In its default mode, git pull is shorthand for git fetch followed by git merge FETCH_HEAD.

위 포스트를 공유드린 이유는 description에서도 볼 수 있다시피 pull은 fetch + merge입니다.
코드를 병합(merge)하고 관리 할 주체가 각 수료생의 repository가 아닌 next-step이기때문에 merge가 아닌 rebase 목적인 걸로 보입니다. 그렇기때문에 한 단계가 끝나면 next-step기준으로 rebase를 하여 맞춰주는 걸로 보입니다.

코드를 병합(merge)하고 관리 할 주체가 각 수료생의 repository가 아닌 next-step이기때문에 단계가 끝나면 next-step기준으로 rebase를 하여 맞춰주는 걸로 보입니다. 그래서 merge가 아닌 rebase를 하기위한 목적으로 보입니다.

위 description 바로 아래에 이런 문구도 있는데
> More precisely, git pull runs git fetch with the given parameters and calls git merge to merge the retrieved branch heads into the current branch. With --rebase, it runs git rebase instead of git merge.

--rebase 옵션을 주면 fetch + rebase가 될 것 같긴 합니다.

Reply:
### 6. 원격 저장소에 잘못된 커밋 메시지를 포함시켜서 푸쉬했을 때, 이를 수정할 수 있는 방법이 있을까요??

[원격 저장소에 올라간 커밋 되돌리기](https://jupiny.com/2019/03/19/revert-commits-in-remote-repository/)

개인(혹은 feature) 브랜치가 아니라면(master, develop) 왠만해서는 수정하지 않는걸 추천드립니다. 만약 수정하신다면 수정하기 전, 후로 팀에 공유해주세요.

Reply:
자세한 답변 감사합니다! 답변주신 내용과 리뷰 코멘트를 리팩터링에 최대한 반영해보겠습니다! 😃